api/controller/sshserver/client.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshserver
+
+import (
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/api/common"
+	apiwatcher "github.com/juju/juju/api/watcher"
+	"github.com/juju/juju/core/watcher"
+	"github.com/juju/juju/rpc/params"
+)
+
+// Client holds the SSH server client for it's respective worker.
+type Client struct {
+	facade base.FacadeCaller
+	*common.ControllerConfigAPI
+}
+
+// NewClient returns an SSH server facade client.
+func NewClient(caller base.APICaller) (*Client, error) {
+	facadeCaller := base.NewFacadeCaller(caller, "SSHServer")
+	return &Client{
+		facade:              facadeCaller,
+		ControllerConfigAPI: common.NewControllerConfig(facadeCaller),
+	}, nil
+}
+
+// WatchControllerConfig provides a watcher for changes on controller config.
+func (c *Client) WatchControllerConfig() (watcher.NotifyWatcher, error) {
+	var result params.NotifyWatchResult
+	if err := c.facade.FacadeCall("WatchControllerConfig", nil, &result); err != nil {
+		return nil, err
+	}
+	if result.Error != nil {
+		return nil, result.Error
+	}
+	return apiwatcher.NewNotifyWatcher(c.facade.RawAPICaller(), result), nil
+}
+
+// SSHServerHostKey returns the private host key for the controller's SSH server.
+func (c *Client) SSHServerHostKey() (string, error) {
+	var result params.SSHServerHostPrivateKeyResult
+	err := c.facade.FacadeCall("SSHServerHostKey", nil, &result)
```

Comment:
Using a StringResult avoids the need for a new bespoke struct. But we've also tended to start introducing bespoke params structs if needed. In this case, given the api semantics, it's unlikely we'll ever want anything other than a string result so we can just use that to minimise the work. We'll be discarding this facade in 4.0 anyway. We'd need to in that case check both the StringResult Error and the facade call err. See existing code examples for how to check a StringResult error.

Reply:
Yup, done, they just check err and return